communication/src/coap_channel.h

```diff
@@ -222,6 +224,11 @@ class __attribute__((packed)) CoAPMessage
 		CoAPType::Enum coapType = CoAP::type(get_data());
 		if (coapType==CoAPType::CON) {
 			timeout = now + transmit_timeout(transmit_count);
+			if (transmit_count == 0) {
+				g_trasmittedMessageCounter++;
```

Comment:
Should we have a bounds check on this to make sure we wont' over flow Int32, in case the user spams the cloud for long enough?

Reply:
Wrap around is fine.

Reply:
Signed integer overflow is undefined behavior. Ideally we should watch out for it, as it will wrap around to large negative numbers instead. Something like this perhaps?

```diff
diff --git a/communication/src/coap_channel.h b/communication/src/coap_channel.h
index d442477ab..8ae212377 100644
--- a/communication/src/coap_channel.h
+++ b/communication/src/coap_channel.h
@@ -25,6 +25,16 @@
 #include "service_debug.h"
 
 #include "communication_diagnostic.h"
+#include <limits>
+
+namespace {
+template <typename T>
+void safeIntegerDiagnosticIncrement(particle::IntegerDiagnosticData<T>& d) {
+       // Cast to underlying
+       auto val = static_cast<typename particle::IntegerDiagnosticData<T>::IntType>(d);
+       d = (val < std::numeric_limits<decltype(val)>::max()) ? val + 1 : 0;
+}
+} // anonymous
 
 namespace particle
 {
@@ -225,9 +235,9 @@ public:
                if (coapType==CoAPType::CON) {
                        timeout = now + transmit_timeout(transmit_count);
                        if (transmit_count == 0) {
-                               g_trasmittedMessageCounter++;
+                               safeIntegerDiagnosticIncrement(g_trasmittedMessageCounter);
                        } else {
-                               g_retransmittedMessageCounter++;
+                               safeIntegerDiagnosticIncrement(g_retransmittedMessageCounter);
                        }
                        transmit_count++;
                        return transmit_count <= MAX_RETRANSMIT+1;
```